Update simulator

AlexLepauvre opened the pull request:
- Make patterns stable over time window
- Change effect specification to enable the specification of effects with specific cross temporal generalization patterns
- Refactorize code to have subfunctions for effects parsing and single subject simulation

AlexLepauvre committed:
Update simulator.py

Update effects specification. Always same number of effects as number of columns in the design matrix

AlexLepauvre committed:
Update core function

Improve flexibility to simulate multivariate patterns with cross temporal generalization

AlexLepauvre committed:
Update tutorials

AlexLepauvre committed:
Add tests

Check all warnings of the core function

AlexLepauvre committed:
Update paper.bib

Add JR king reference

AlexLepauvre committed:
Update conf.py

BF: missing coma

AlexLepauvre committed:
Update pyproject.toml

AlexLepauvre committed:
Update pyproject.toml

qian-chu commented:
I like the refactorization! A few suggestions to consider:

- It might make sense to reorder the input arguments to `Simulator(X, effects, ...)`, as the design matrix and effects are closely related to each other.
- `generate_events` could be simplified, as `condition_names` can now be retrieved from the DataFrame `X`.
- `summary` method could be replaced by `__repr__`

Several bugs I noticed:
- Mathematical expressions in `Simulator` docstr and tutorials are not rendered correctly
- docs\source\tutorial\index.rst should update tutorial names, namely:

```
.. toctree::
   :maxdepth: 2

   01-multisim_basic
   02-multisim_kernel
   03-multisim_intersubject_noise
   04-multisim_cross_temporal_generalization
   05-multisim_effect_size
```

AlexLepauvre committed:
Update simulator.py

Normalize pattern to be of Mahalanobis distance 1 and not Euclidean distance 1

AlexLepauvre committed:
Update simulator.py

Fix math formulae and simplify generate_events

AlexLepauvre committed:
Update tutorials

Add a separate tutorial for mathematical details instead of having it in the paper

AlexLepauvre committed:
Update simulator.py

AlexLepauvre committed:
Update paper and citations

AlexLepauvre committed:
Update paper.md

AlexLepauvre committed:
Update paper

AlexLepauvre committed:
(auto) Paper PDF Draft

AlexLepauvre committed:
Update paper.md

AlexLepauvre committed:
(auto) Paper PDF Draft

AlexLepauvre committed:
Update conf.py

Add mathjax for parsing math formulae

AlexLepauvre committed:
Update conf.py

AlexLepauvre committed:
Update docs

Update documentation

AlexLepauvre committed:
(auto) Paper PDF Draft

AlexLepauvre committed:
Small changes

AlexLepauvre committed:
BF: citations

AlexLepauvre commented:
All suggestions have been taken into account. Added a notebook describing all the maths
